src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt

```diff
@@ -14,57 +14,55 @@ import java.io.File
 
 data class RustToolchain(val location: String) {
 
-    private fun hasExecuteable(exec: String): Boolean =
-        File(pathToExecutable(exec)).canExecute()
-
     fun looksLikeValidToolchain(): Boolean =
-        hasExecuteable(CARGO)
-
-    fun containsMetadataCommand(): Boolean {
-        check(!ApplicationManager.getApplication().isDispatchThread)
-
-        val cmd = nonProjectCargo().generalCommand("metadata", listOf("--help"))
-        val procOut = try {
-            CapturingProcessHandler(cmd.createProcess(), Charsets.UTF_8, cmd.commandLineString).runProcess(1.seconds)
-        } catch (e: ExecutionException) {
-            return false
-        }
-        return procOut.exitCode == 0
-    }
+        hasExecutable(CARGO)
 
-    fun queryCargoVersion(): Version? {
+    fun queryVersions(): VersionInfo {
         check(!ApplicationManager.getApplication().isDispatchThread)
-        if (!looksLikeValidToolchain()) return null
+        if (!looksLikeValidToolchain()) return VersionInfo(null, null, null, false)
```

Comment:
Urgh, apparently old IDEA does not have `SemVer.UNKNOWN`, so it's simpler to resort to `null`s for the time being.